sonar-core/src/main/java/org/sonar/core/component/ComponentKeys.java

```diff
@@ -30,7 +30,7 @@ public final class ComponentKeys {
   /**
    * Should be in sync with DefaultIndexedFile.MAX_KEY_LENGTH
    */
-  public static final int MAX_COMPONENT_KEY_LENGTH = 400;
+  public static final int MAX_COMPONENT_KEY_LENGTH = 255;
```

Comment:
Keep it 400 otherwise we also have to do it in DefaultIndexedFile as mentioned in above comment. Lets keep it 400 at backend